### PR TITLE
[Hub Generated] Review request for Microsoft.Media/Encoding to add version stable/2022-07-01

### PR DESCRIPTION
**Pull request by jinshang-msft**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/Chuansssss/azure-rest-api-specs/mediaservices-Microsoft.Media-2022-07-01...main/).
## ARM API Information (Control Plane)

Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by rkmanda** [comment]
>         "experimentalOptions": {

Are these user defined key value pairs?

---
Refers to: specification/mediaservices/resource-manager/Microsoft.Media/Encoding/stable/2022-07-01/Encoding.json:1546 in 78b7e5a. [](commit_id = 78b7e5a8324fbae41d02872bbdf7d0403cfcf1da, deletion_comment = False)

**Comment by Chuansssss** [comment]
> > ```
> >     "experimentalOptions": {
> > ```
> 
> Are these user defined key value pairs?
> 
> Refers to: specification/mediaservices/resource-manager/Microsoft.Media/Encoding/stable/2022-07-01/Encoding.json:1546 in [78b7e5a](https://github.com/Azure/azure-rest-api-specs/commit/78b7e5a8324fbae41d02872bbdf7d0403cfcf1da). [](commit_id = [78b7e5a](https://github.com/Azure/azure-rest-api-specs/commit/78b7e5a8324fbae41d02872bbdf7d0403cfcf1da), deletion_comment = False)

It is handled as ODataTypedValueDictionary<string> on the server side. Due to the limitation of our internal API framework, this type cannot be handled correctly. We have been defining it as string and providing guidance in the description in the previous API version. This one follows the pattern again.

**Comment by rkmanda** [comment]
>         "experimentalOptions": {

Per offline discussion. These are user defined keys and the service doesnt interpret the set of keys. Its modelled as a string due to the dependency on a different team that cannot provide support for the dictionary.

---
In reply to: [1286400514](https://github.com/Azure/azure-rest-api-specs/pull/21190#issuecomment-1286400514) [](http://example.com/codeflow?ancestors=1286400514,1286400514)

---
Refers to: specification/mediaservices/resource-manager/Microsoft.Media/Encoding/stable/2022-07-01/Encoding.json:1546 in 78b7e5a. [](commit_id = 78b7e5a8324fbae41d02872bbdf7d0403cfcf1da, deletion_comment = False)

**Comment by rkmanda** [comment]
>         "experimentalOptions": {

suggested releasing this as a preview API version and modify the behavior of this property before going GA

---
In reply to: [1306380963](https://github.com/Azure/azure-rest-api-specs/pull/21190#issuecomment-1306380963) [](http://example.com/codeflow?ancestors=1306380963,1286400514,1286400514)

---
Refers to: specification/mediaservices/resource-manager/Microsoft.Media/Encoding/stable/2022-07-01/Encoding.json:1546 in 78b7e5a. [](commit_id = 78b7e5a8324fbae41d02872bbdf7d0403cfcf1da, deletion_comment = False)

File: specification/mediaservices/resource-manager/Microsoft.Media/Encoding/stable/2022-07-01/Encoding.json [hunk 1]
```diff
@@ -0,0 +1,3294 @@
...
+      "type": "string",
+      "description": "The Media Services account name.",
+      "x-ms-parameter-location": "method"
+    },
+    "ApiVersion": {
+      "name": "api-version",
+      "in": "query",
+      "required": true,
+      "type": "string",
+      "description": "The Version of the API to be used with the client request."
```
Comment (giakas) [inline specification/mediaservices/resource-manager/Microsoft.Media/Encoding/stable/2022-07-01/Encoding.json:3291]: Looks like the word version was changed to Version, please revert the casing change, this string needs to match all service swagger files

**Review by giakas — APPROVED** [review]

**Comment by akning-ms** [comment]
@qiaozha to approve JS breaking change
@fengzhou-msft to check why .Net SDK track 2 generation failure
@jianyexi  to check why SuppressionReviewRequired

**Comment by fengzhou-msft** [comment]
The `azure-sdk-for-net-track2` failure is caused by the `Fade.Duration` property not having a format which is a specific requirement by .NET. It could be fixed in the `autorest.md` of `azure-sdk-for-net` later.

**Comment by giakas** [comment]
> The `azure-sdk-for-net-track2` failure is caused by the `Fade.Duration` property not having a format which is a specific requirement by .NET. It could be fixed in the `autorest.md` of `azure-sdk-for-net` later.

@fengzhou-msft I think we should fix it in the PR. @Chuansssss can you add "format": "duration" to duration property. I don't have access to make changes to your fork.

**Pull request merged**